test/api/proxy.spec.js

```diff
@@ -1,14 +1,40 @@
 const chai = require('chai');
 const chaiHttp = require('chai-http');
-require('../../src/server');
-
 const { expect } = chai;
 chai.use(chaiHttp);
 const remoteServer = 'http://127.0.0.1';
 const port = process.env.PORT || 4000;
+process.env.DELAY_IN_SECONDS = 1.6;
+process.env.EXPIRE_IN_SECONDS = 3;
+process.env.REQUEST_LIMIT = 1;
+process.env.BACK_END_URL = '';
+require('../../src/server');
 
 describe('reli-proxy', function module1() {
   this.timeout(10000);
+  /*
+  * Test the throttling
+  */
+  describe('/Testing the throttling of requests', () => {
+    it('it should rate limit second request', (done) => {
```

Comment:
nice test

Reply:
@michalmikolajczyk yes, the Redis store is required for both speed limiting and rate-limiting as both the tasks require some kind of memory storage for keeping track of requests for IP, by default these module use MemoryStore.
1. Regarding the changes required, do you need a separate test just for rate limiting as the current test added shows both throttling with retry and rate limiting?
2. Currently, throttling is done by IP taken from `req.ip` and the same is stored as key in Redis store. Is some other key required here?
Thanks :)